src/SFA.DAS.EAS.Web/Validators/ApprenticeshipCoreValidator.cs

```diff
@@ -71,7 +79,9 @@ protected virtual void ValidateUln()
         protected virtual void ValidateTraining()
         {
             RuleFor(x => x.TrainingCode)
-                .NotEmpty().WithMessage(ValidationText.TrainingCode01.Text).WithErrorCode(ValidationText.TrainingCode01.ErrorCode); ;
+                .Cascade(CascadeMode.StopOnFirstFailure)
+                .NotEmpty().WithMessage(ValidationText.TrainingCode01.Text)
```

Comment:
It would be nice to give the existing ValidationMessages meaningful names as we touch them.
E.g. we could rename TrainingCode01 -> TrainingCodeCantBeEmpty.

Reply:
I accidentally chopped this line so didn't really touch it. The poor naming I believe was deliberate - each type of error links back to some validation rule defined by the business and documented some place. However, it may be that the naming is no longer worth keeping, and it is unlikely that the documentation has been kept up to date with rules we've since added.